orangewidget/settings.py

```diff
@@ -381,63 +381,58 @@ def store_initialization_data(self, initialization_data: dict) -> None:
         """
         self.initialization_data = initialization_data
 
-    @staticmethod
-    def _default_packer(setting, instance):
-        """A simple packet that yields setting name and value.
+    @classmethod
+    def default_packer(cls,
+                       setting: Setting,
+                       component: OWComponent,
+                       handler: "SettingsHandler") -> Iterator[Tuple[str, Any]]:
+        """Yield setting name and value for packable (= non-context) setting."""
+        if setting.packable:
+            value = getattr(component, setting.name)
+            handler.check_warn_type(value, setting, component)
+            yield setting.name, value
+
+    PackerType = Callable[[Setting, OWComponent, "SettingsHandler"],
+                          Iterator[Tuple[str, Any]]]
 
-        Parameters
-        ----------
-        setting : Setting
-        instance : OWBaseWidget
+    def pack(self, widget: "OWBaseWidget",
+             packer: Optional[PackerType] = None) -> dict:
         """
-        if setting.packable:
-            if hasattr(instance, setting.name):
-                yield setting.name, getattr(instance, setting.name)
-            else:
-                warnings.warn("{0} is declared as setting on {1} "
-                              "but not present on instance."
-                              .format(setting.name, instance))
```

Comment:
I removed this warning because I don't see a situation in which it can (reasonably) happen: how can a setting appear in a widget without being declared? Unless, of course, one adds it "manually" to the dict, or if it deletes it using delattr.